microsoft/testsuites/osu/osusuite.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import re
+import os
+from pathlib import Path
+
+from assertpy import assert_that
+
+from lisa import (
+    LisaException,
+    Logger,
+    Node,
+    SkippedException,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    constants,
+    simple_requirement,
+    CustomScript,
+    CustomScriptBuilder
+)
+from lisa.features import Gpu, GpuEnabled, SerialConsole
+from lisa.features.gpu import ComputeSDK
+from lisa.operating_system import Debian
+from lisa.tools import Lspci, NvidiaSmi, Pip, Python, Reboot, Service, Tar, Wget
+from lisa.util import get_matched_str
+
+_osu_mpi_location = ("https://mvapich.cse.ohio-state.edu/download/mvapich/mv2/mvapich2-2.3.7-1.tar.gz")
+_osu_mpi_file_name = "mvapich2.tgz"
+
+_osu_bench_location = ("http://mvapich.cse.ohio-state.edu/download/mvapich/osu-micro-benchmarks-6.2.tar.gz")
+_osu_bench_file_name = "osu-bench.tgz"
+
+
+@TestSuiteMetadata(
+    area="hpc",
+    category="functional",
+    name="OSU Bench",
+    description="""
+    This test suite runs the OSU Micro-Benchmarks MPI test cases.
+    """,
+)
+class OSUTestSuite(TestSuite):
+    TIMEOUT = 200000
+
+    @TestCaseMetadata(
+        description="""
+            This test case runs GPU/CPU MPI latency.
+
+            Steps:
+            1. Install MVAPICH;
+            2. Install OSU Micro_Benchmarks;
+            3. Run GPU/CPU collective/latency tests in a single node.
+
+        """,
+        timeout=TIMEOUT,
+        requirement=simple_requirement(
+            supported_features=[GpuEnabled(), SerialConsole],
+        ),
+        priority=1,
+    )
+    def run_osu_bench(self, node: Node, log_path: Path, log: Logger) -> None:
+
+        _install_osu_mpi(node)
+        path = _install_osu_bench(node)
+
+        gpu_count = node.capability.gpu_count
+        gpu = node.features[Gpu]
+        expected_count = gpu.get_gpu_count_with_lspci()
+        assert_that(gpu_count).described_as(
+            "Expected device count didn't match actual device count from lspci!"
+        ).is_equal_to(expected_count)
+
+        '''test_script = CustomScriptBuilder(Path(__file__).parent.joinpath("."), ["install_dependencies_alma.sh"])
+        script: CustomScript = node.tools[test_script]
+        result = script.run("alma8.6")'''
+        node.execute(
+            f"mpirun -np {gpu_count} python3 {path}/python/run.py --benchmark allgather --min 128 --max 10000000 >> /tmp/osu_bench.log", #--buffer pycuda #--buffer cupy
```

Comment:
Appending to a log file will result in progress not being displayed on the lisa logs. Is this file being used somewhere? Also, please make a tools for `mpirun`

Reply:
I used this log file for debugging purposes. Removed.